app/models/mixins/relationship_mixin.rb

```diff
@@ -580,9 +580,9 @@ def child_types(*args)
   end
 
   def add_parent(parent)
-    return update(:parent => parent) if use_ancestry?
-
-    parent.with_relationship_type(relationship_type) { parent.add_child(self) }
+    parent.with_relationship_type(relationship_type) do
+      use_ancestry? ? update(:parent => parent) : parent.add_child(self)
```

Comment:
could you use `if` instead of ternary?
I think of ternary as focusing on the return vs focusing on the flow